.github/workflows/ci.yml

```diff
@@ -44,7 +48,10 @@ jobs:
     - name: Build
       run: |
         cabal v2-update
-        cabal v2-build --enable-tests --enable-benchmarks
+        cabal v2-configure \
+          --constraint 'Cabal-syntax installed' \
+          --enable-tests --enable-benchmarks
```

Comment:
`haskell-actions/setup` already installs Cabal from the matrix.

the added `--constraint 'Cabal-syntax installed'` prevents the build from satisfying the dependency via compiling another (newest) version.